lib/DelegatedModule.js

```diff
@@ -2,79 +2,83 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var Module = require("./Module");
-var OriginalSource = require("webpack-sources").OriginalSource;
-var RawSource = require("webpack-sources").RawSource;
-var WebpackMissingModule = require("./dependencies/WebpackMissingModule");
-var DelegatedSourceDependency = require("./dependencies/DelegatedSourceDependency");
+"use strict";
 
-function DelegatedModule(sourceRequest, data, type, userRequest) {
-	Module.call(this);
-	this.sourceRequest = sourceRequest;
-	this.request = data.id;
-	this.meta = data.meta;
-	this.type = type;
-	this.userRequest = userRequest;
-	this.built = false;
-	this.delegateData = data;
-}
-module.exports = DelegatedModule;
+const Module = require("./Module");
+const OriginalSource = require("webpack-sources").OriginalSource;
+const RawSource = require("webpack-sources").RawSource;
+const WebpackMissingModule = require("./dependencies/WebpackMissingModule");
+const DelegatedSourceDependency = require("./dependencies/DelegatedSourceDependency");
+
+class DelegatedModule extends Module {
+	constructor(sourceRequest, data, type, userRequest) {
+		super();
+		this.sourceRequest = sourceRequest;
+		this.request = data.id;
+		this.meta = data.meta;
+		this.type = type;
+		this.userRequest = userRequest;
+		this.built = false;
+		this.delegated = true;
```

Comment:
`delegated` is not in the original source.

Why adding it? If there is a reason, could you move it into a separate PR. The "refactor to ES6" PRs should only refactor to ES6...